spyder/plugins/maininterpreter.py

```diff
@@ -161,18 +161,17 @@ def python_executable_switched(self, custom):
     def is_python_interpreter(self, pyexec):
         if not osp.isfile(pyexec):
             return False
-        args = ["-c", "print('valid_python_interpreter')"]
         try:
-            proc = programs.run_program(pyexec, args)
+            proc = programs.run_program(pyexec, ["-h"])
             valid = str(proc.communicate()[0])
-            if not valid.startswith("b'valid_python_interpreter"):
+            if len(valid) > 2 and valid[2:].startswith("usage: {}".format(pyexec)):

```

Comment:
I think this match is still too weak. Please create a more complete function with some of my previous suggestions, add it to utils/programs.py and also add a test for it in utils/tests/test_programs.py